examples/test/action_server_tests.cpp

```diff
@@ -0,0 +1,140 @@
+  float expected_distance =
+    std::sqrt(goal->target_x * goal->target_x + goal->target_y * goal->target_y);
+  EXPECT_GT(expected_distance, 10.0);
+
+  auto response = server_mock->call_real_handle_goal(uuid, goal);
+
+  /// Verify the goal was rejected due to distance
+  EXPECT_EQ(response, rclcpp_action::GoalResponse::REJECT);
+}
+
+TEST_F(ActionServerTest, FeedbackPublishingWithValidData)
+{
+  auto node = std::make_shared<test_composition::ActionServer>(opts);
+  auto server_mock =
+    rtest::experimental::findActionServer<rtest_examples_interfaces::action::MoveRobot>(
+      node, "move_robot");
+  ASSERT_TRUE(server_mock);
+
+  /// Test data: goal at (3.0, 4.0) - distance = 5.0
+  auto goal = std::make_shared<rtest_examples_interfaces::action::MoveRobot::Goal>();
+  goal->target_x = 3.0;
+  goal->target_y = 4.0;
+
+  auto mock_goal_handle =
+    rtest::experimental::createMockGoalHandle<rtest_examples_interfaces::action::MoveRobot>(goal);
+
+  /// Verify initial position
+  auto [initial_x, initial_y] = node->get_current_position();
+  EXPECT_FLOAT_EQ(initial_x, 0.0);
+  EXPECT_FLOAT_EQ(initial_y, 0.0);
+
+  /// Set up expectation: publish_feedback should be called with reasonable values
+  EXPECT_CALL(
+    *mock_goal_handle,
+    publish_feedback(::testing::AllOf(
+      ::testing::Pointee(::testing::Field(
+        &rtest_examples_interfaces::action::MoveRobot::Feedback::current_x,
+        ::testing::AllOf(::testing::Ge(0.0), ::testing::Le(3.0)))),
+      ::testing::Pointee(::testing::Field(
+        &rtest_examples_interfaces::action::MoveRobot::Feedback::current_y,
+        ::testing::AllOf(::testing::Ge(0.0), ::testing::Le(4.0)))),
+      ::testing::Pointee(::testing::Field(
+        &rtest_examples_interfaces::action::MoveRobot::Feedback::distance_remaining,
+        ::testing::AllOf(::testing::Ge(0.0), ::testing::Le(5.0)))))))
+    .Times(5)
+    .WillRepeatedly(::testing::Invoke(
+      [goal](std::shared_ptr<rtest_examples_interfaces::action::MoveRobot::Feedback> feedback) {
+        std::cout << "Feedback step: position(" << feedback->current_x << ", "
+                  << feedback->current_y << ") distance_remaining: " << feedback->distance_remaining
+                  << " target(" << goal->target_x << ", " << goal->target_y << ")" << std::endl;
+
+        /// Verify distance_remaining calculation is correct
+        float expected_distance_remaining = std::sqrt(
+          (goal->target_x - feedback->current_x) * (goal->target_x - feedback->current_x) +
+          (goal->target_y - feedback->current_y) * (goal->target_y - feedback->current_y));
+        EXPECT_NEAR(feedback->distance_remaining, expected_distance_remaining, 0.01);
+
+        /// Verify we're moving away from origin (progress check)
+        float distance_from_origin = std::sqrt(
+          feedback->current_x * feedback->current_x + feedback->current_y * feedback->current_y);
+        EXPECT_GT(distance_from_origin, 0.0);
+      }));
```

Comment:
That looks scary and complicated. But I have no better idea how to simplify it atm.
I guess it's good for first experimental release :)